torchmetrics/text/eed.py

```diff
@@ -0,0 +1,141 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, List, Optional, Sequence, Tuple, Union
+
+from torch import Tensor, stack
+from typing_extensions import Literal
+
+from torchmetrics.functional.text.eed import _eed_compute, _eed_update
+from torchmetrics.metric import Metric
+
+
+class EED(Metric):
+    """Computes extended edit distance score (`EED`_) [1] for strings or list of strings. The metric utilises the
+    Levenshtein distance and extends it by adding an additional jump operation.
+
+    Args:
+        language:
+            Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+        return_sentence_level_score:
+            An indication of whether sentence-level EED is to be returned
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
+        rho:
+            coverage cost, penalty for repetition of characters
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+        compute_on_step:
+            Forward only calls ``update()`` and return None if this is set to False.
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the allgather
+
+    Return:
+        Extended edit distance score as a tensor
+
+    Example:
+        >>> from torchmetrics.text import EED
+        >>> reference_corpus = ["this is the reference", "here is another one"]
+        >>> hypothesis_corpus = ["this is the prediction", "here is an other sample"]
+        >>> metric = EED()
+        >>> metric(reference_corpus=reference_corpus, hypothesis_corpus=hypothesis_corpus)
+        tensor(0.3204)
+
+    References:
+        [1] P. Stanchev, W. Wang, and H. Ney, “EED: Extended Edit Distance Measure for Machine Translation”, submitted
+        to WMT 2019. `EED`_
+    """
+
+    sentence_eed: List[Tensor] = []
+    higher_is_better: False
```

Comment:
```suggestion
    higher_is_better = False
```
To fix `torch.jit.script`

Reply:
thanks!